Model/src/main/java/org/gusdb/wdk/model/answer/AnswerValue.java

```diff
@@ -906,37 +907,67 @@ private String getPartitionKeysString(String queryName) throws WdkModelException
       RecordClass rc = _question.getRecordClass();
       SqlQuery partKeySqlQuery = rc.getPartitionKeysSqlQuery();
 
+      PrimaryKeyDefinition pkd = rc.getPrimaryKeyDefinition();
+      String idSql = getIdSql();
+      String partSql = partKeySqlQuery.getSql();
+
+      String sql =
+          "SELECT distinct partition_key " +
+          " FROM (" + idSql + ") ids, (" + partSql + ") parts" +
+          " WHERE " + pkd.createJoinClause("ids", "parts");
+      return getPartitionKeysString(rc, queryName, sql);
```

Comment:
You don't set _partitionKeysString here any more, so you will always fall into the if.